### PR TITLE
Add tests and custom strategies

### PR DESCRIPTION
**Pull request by odarriba**
Solving #17 .

Main library functionality is now tested against Travis CI.

Also, a bonus functionality has been aded: **custom strategies**

That feature allow end user to define custom strategies and use them inside Ravenx if it meets the strategy behaviour defined in the library.

Also, a `Dummy` strategy has been added to the library, as it seems useful when starting to use Ravenx or just for debugging purposes.

Oh! and this seems like the final PR of the 1.0.0 release, so `README.md`, `CHANGELOG.md` and `mix.exs` has been updated.

❤️

### CODE REVIEW

**Comment by enoliglesias** [comment]
:heart: :green_heart: :yellow_heart: :blue_heart: :purple_heart:

File: lib/ravenx/notification.ex [hunk 1]
```diff
@@ -27,7 +27,7 @@ defmodule Ravenx.Notification do
       def dispatch(opts) do
         opts
         |> get_notifications_config
-        |> Enum.map(fn(k, opts) -> {k, Ravenx.Notification.dispatch_notification(opts)} end)
+        |> Enum.map(fn({k, opts}) -> {k, Ravenx.Notification.dispatch_notification(opts)} end)
```
Comment (rsierra) [inline lib/ravenx/notification.ex]: Function captures ¿Hot or not?
|> Enum.map(&({&1, Ravenx.Notification.dispatch_notification(&2)}))

File: test/support/test_config.ex [hunk 2]
```diff
@@ -0,0 +1,14 @@
+defmodule Ravenx.Test.TestConfig do
+
+  def test_multiple (payload) do
```
Comment (rsierra) [inline test/support/test_config.ex]: It doesn't affect anything, but payload is never used and there is an space between function name and parenthesis (in the next function too).

File: lib/ravenx/notification.ex (hunk 1 above)
Comment (odarriba) [inline lib/ravenx/notification.ex]: Totally hot

File: test/support/test_config.ex (hunk 2 above)
Comment (odarriba) [inline test/support/test_config.ex]: Fixed!

File: lib/ravenx/notification.ex (hunk 1 above)
Comment (rsierra) [inline lib/ravenx/notification.ex]: Yeah, I missed that detail 😅

**Pull request merged**